### PR TITLE
Fix conditional operator after typeof

### PR DESCRIPTION
**Pull request by singalhimanshu**
The following code should not produce a warning (E057):
     'undefined' === typeof foo ? 3 : 4;

Fix: #306

### CODE REVIEW

**Comment by singalhimanshu** [comment]
Let me know if this fix was okay. If so I will write the required test cases for this condition.

File: src/quick-lint-js/parse.h [hunk 1]
```diff
@@ -741,9 +741,18 @@ class parser {
     }
     case expression_kind::_typeof: {
       expression *child = ast->child_0();
-      if (child->kind() == expression_kind::variable) {
+      switch (child->kind()) {
+      case expression_kind::conditional: {
+        if (child->child_0()->kind() == expression_kind::variable) {
```
Comment (strager) [inline src/quick-lint-js/parse.h:746]: Blocking: This causes false negatives for other cases, like `"undefined" === typeof foo.x ? 3 : 4;`.

**Review by strager — CHANGES_REQUESTED** [review]
Blocking: I think editing `visit_expression` is the wrong approach. We should fix the parsing.

> If so I will write the required test cases for this condition.

We'd want the test cases regardless of what the solution is, no?

File: src/quick-lint-js/parse.h (hunk 1 above)
Comment (singalhimanshu) [reply src/quick-lint-js/parse.h]: > We'd want the test cases regardless of what the solution is, no?

Yes

**Comment by singalhimanshu** [comment]
> We'd want the test cases regardless of what the solution is, no?

Yes =S

**Review by strager — APPROVED** [review]

**Pull request merged**